core/src/pixel_bender.rs

```diff
@@ -367,11 +379,30 @@ fn read_op<R: Read>(
                     .push(Operation::LoadFloat { dst, mask, val })
             }
         }
+        Opcode::SampleNearest | Opcode::SampleLinear => {
+            let dst = data.read_u16::<LittleEndian>()?;
+            let mask = data.read_u8()?;
+            let src = read_uint24(data)?;
+            let tf = data.read_u8()?;
+            match opcode {
+                Opcode::SampleNearest => {
+                    shader
+                        .operations
+                        .push(Operation::SampleNearest { dst, mask, src, tf })
+                }
+                Opcode::SampleLinear => {
+                    shader
+                        .operations
+                        .push(Operation::SampleLinear { dst, mask, src, tf })
+                }
+                _ => unreachable!(),
+            }
+        }
         _ => {
             let dst = data.read_u16::<LittleEndian>()?;
             let mask = data.read_u8()?;
             let src = read_uint24(data)?;
-            assert_eq!(data.read_u8()?, 0);
+            assert_eq!(data.read_u8()?, 0, "Unepxected u8 for opcode {opcode:?}");
```

Comment:
If this is possible in case of a corrupt shader, would it be a good idea to move this behind a warn/err?

Reply:
This caught a parsing bug, so I want to keep it in order to encourage bug reports

Reply:
W.R.T. the same line, there's a typo: "Unepxected"